rmf_fleet_adapter/src/full_control/main.cpp

```diff
@@ -1018,6 +1023,35 @@ std::shared_ptr<Connections> make_fleet(
       connections->closed_lanes_pub->publish(state_msg);
     });
 
+  connections->speed_limit_request_sub =
+    adapter->node()->create_subscription<
+    rmf_fleet_msgs::msg::SpeedLimitRequest>(
+    rmf_fleet_adapter::SpeedLimitRequestTopicName,
+    rclcpp::SystemDefaultsQoS(),
+    [w = connections->weak_from_this(), fleet_name](
+      rmf_fleet_msgs::msg::SpeedLimitRequest::ConstSharedPtr request_msg)
+    {
+      const auto connections = w.lock();
+      if (!connections)
+        return;
+
+      if (request_msg->fleet_name != fleet_name &&
+      !request_msg->fleet_name.empty())
+        return;
```

Comment:
From what I see the behavior is that either `fleet_name` is specified or the same request is applied to all the fleets.
This sounds sensible but I'm a bit worried about the utility and possible unexpected behavior of "apply to all fleets" with the current definition of lane closure / speed limit.

Lane closures are issued based on the navgraph index but there is no guarantee that they mean anything similar among different navgraphs. For example, issuing a request with no fleet_name to block lane 0 in the airport terminal world ends up closing a charger lane for CleanerBot, a junction in the corridor for tinyRobot and something else entirely for deliveryRobot.

Imo it would be better to either change the definition of a lane closure (for example between named waypoints only and enforcing waypoints to have names, or by coordinates although that is quite tricky) or remove this behavior altogether and make a message with an empty `fleet_name` an error / noop. I can't really imagine a case in which a user would want to close a lane among multiple fleets, specifying an ID that has a completely different meaning for each fleet.

Does this make sense?

Reply:
Fix https://github.com/open-rmf/rmf_ros2/pull/217/commits/c1d27cf6b8f6cb7f44cea97df501bcb129b32a81